src/main/java/flight/reservation/payment/Payment.java

```diff
@@ -0,0 +1,37 @@
+package flight.reservation.payment;
+
+public class Payment {
+    private static final String PAYMENT_ERROR_MESSAGE = "Payment information is not set or not valid.";
+
+    private static boolean cardIsPresentAndValid(CreditCard card) {
+        return card != null && card.isValid();
+    }
+
+    public static boolean payWithCreditCard(CreditCard card, double amount) throws IllegalStateException {
```

Comment:
maybe introduce a PaymentMethod interface which is implemented by CreditCard and Paypal and move the implementations (payWith... to the CreditCard class or Paypal class)